core/commands/show_command.rb

```diff
@@ -81,7 +81,10 @@ def execute
 
   def show_private_ip_address(params)
     Configuration.from_spec(params.first).and_then do |config|
-      network_settings = NetworkSettings.from_file(config.network_settings_file)
+      NetworkSettings.from_file(config.network_settings_file).and_then do |network_settings|
```

Comment:
We call the same code 3 times in this method. Please add a method that will get the configuration and network settings object and use it in all applicable situations.